src/riot-rs-threads/src/arch/cortex_m.rs

```diff
@@ -216,53 +212,42 @@ unsafe extern "C" fn PendSV() {
 /// This function is called in PendSV.
 // TODO: make arch independent, or move to arch
 #[no_mangle]
-unsafe fn sched(old_sp: usize) -> usize {
-    let cs = CriticalSection::new();
-    let next_pid;
-
+unsafe fn sched(old_sp: usize) -> u128 {
     loop {
-        {
-            if let Some(pid) = (&*THREADS.as_ptr(cs)).runqueue.get_next() {
-                next_pid = pid;
-                break;
+        if let Some(res) = critical_section::with(|cs| {
+            let threads = &mut *THREADS.as_ptr(cs);
+            if let Some(next_pid) = threads.runqueue.get_next() {
```

Comment:
Nit: You could use match here and use an early return in case of `None` to make the code more readable.

Reply:
how do you mean?

Reply:
Like this: 
```rs
   let next_id = match threads.runqueue.get_next() {
        Some(pid) => pid,
        None => {
            cortex_m::asm::wfi();
            return None

        }
    };

    let current_high_regs = ...;

    ...

    Some(res)
```

Not a huge diff, but would get rid of one nesting layer.

Reply:
yup